bittensor/utils/__init__.py

```diff
@@ -449,6 +450,18 @@ def millify(n: int):
 
     return '{:.0f}{}'.format(n / 10**(3 * millidx), millnames[millidx])
 
+@backoff.on_exception(backoff.constant,
+                            Exception,
+                            interval=1,
+                            max_tries=3)
+def get_block_with_retry(subtensor: 'bittensor.Subtensor') -> Tuple[int, int, bytes]:
+    block_number = subtensor.get_current_block()
+    difficulty = subtensor.difficulty
+    block_hash = subtensor.substrate.get_block_hash( block_number )
+    if block_hash is None:
+        raise Exception("Network error. Could not connect to substrate to get block hash")
+    return block_number, difficulty, block_hash
```

Comment:
I am missing the happy case test when we have no exception. Do we have it?

Reply:
btw, good movement extracting this.

Reply:
added some success tests for this